client/components/test-mode-notice/index.tsx

```diff
@@ -0,0 +1,180 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { getPaymentSettingsUrl, isInTestMode } from 'utils';
+import BannerNotice from '../banner-notice';
+import interpolateComponents from '@automattic/interpolate-components';
+import { Link } from '@woocommerce/components';
+
+type CurrentPage =
+	| 'overview'
+	| 'documents'
+	| 'deposits'
+	| 'disputes'
+	| 'loans'
+	| 'payments'
+	| 'transactions';
+
+interface Props {
+	currentPage: CurrentPage;
+	actions?: ReadonlyArray< {
+		label: string;
+		className?: string;
+		variant?: Button.Props[ 'variant' ];
+		url?: string;
+		onClick?: React.MouseEventHandler< HTMLAnchorElement >;
+	} >;
```

Comment:
You can also reuse component type definitions with something like:
```tsx
React.ComponentProps< typeof BannerNotice >[ 'actions' ]
```

Reply:
Great tip! Updated in 02a03fe71